### PR TITLE
Add new word search program

### PR DESCRIPTION
**Pull request by fplouis**
@andrew-qa @burkestar 

This is the pull request for my word search program. 

It will generate a word search of size 15 by default, or any size passed in with -s up to 100, and then solve it for all words in words.txt and print the results. Repeat words can optionally be included.

There is a unit test that tests the functionality of the word checking method individually, using a predefined grid instead of a randomly generated one, so we can check the expected results.

### CODE REVIEW

File: test_wordsearch.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class TestWordsearch(unittest.TestCase):
+
+    def test_word_check(self):
+        # set up an instance of the wordsearch controller
+        grid_size = 5
+        ws = wordsearch_controller.WordSearchController(size=grid_size, duplicate=True)
+        ws.read_words_from_file()
+        ws.generate_grid()
```
Comment (andrew-qa) [inline test_wordsearch.py:12]: I'm not sure why do we need to generate grid if we overriding it later?

Comment (fplouis) [reply test_wordsearch.py:12]: The generate_grid method also defines the size of both grids, which we needed before being able to overwrite them. That could be done either by calling the generate_grid method like I did here, or by just copying out the logic that does it and placing it here instead.

Comment (andrew-qa) [reply test_wordsearch.py:12]: Got it, thank you for explanation

Comment (fplouis) [reply test_wordsearch.py:12]: Is there anything else you'd like to know about this project?

**Pull request merged**